src/bfd_resolve.cpp

```diff
@@ -56,29 +56,25 @@ static boost::filesystem::path check_path(const std::string& name)
 
 Initializer Lib::dummy_;
 
-Lib::Lib(const std::string& name) : name_(name)
+Lib::Lib(const std::string& name)
+: name_(name), handle_(bfd_openr(check_path(name).c_str(), nullptr))
 {
-    handle_ = bfd_openr(check_path(name).c_str(), nullptr);
-    if (handle_ == nullptr)
+    if (!handle_)
     {
         throw InitError("failed to open BFD handle", name);
     }
     // init stuff
     //  Not sure if those are needed for side-effects
-    bfd_get_arch_size(handle_);
-    bfd_find_target(nullptr, handle_);
-    if (!bfd_check_format(handle_, bfd_object))
+    bfd_get_arch_size(handle_.get());
+    bfd_find_target(nullptr, handle_.get());
+    if (!bfd_check_format(handle_.get(), bfd_object))
     {
-        bfd_close(handle_);
-        handle_ = nullptr;
         throw InitError("Not a valid bfd_object", name);
     }
 
     char** matching;
-    if (bfd_check_format_matches(handle_, bfd_archive, &matching))
+    if (bfd_check_format_matches(handle_.get(), bfd_archive, &matching))
```

Comment:
According to [this](https://ftp.gnu.org/old-gnu/Manuals/bfd-2.9.1/html_node/bfd_22.html#SEC76), we are leaking memory here. Money quote: 
> When done with the list that `matching` points to, the caller should free it.

As we don't look at the`matching`, why not pass nullptr? Heck, why not just use `bfd_check_format`?

@tilsche

Reply:
I guess we should just use `bfd_check_format` here though I'm not quite sure why we need this check at all...

Reply:
I've gone ahead and removed  that check all together because we will never end up in that codepath anyways with the other bfd_check_format some lines above